File: server-scala-play-server/app/model/Vehicle.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for Vehicle.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class Vehicle(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  name: Option[String],
  description: Option[String],
  vehicleType: Option[VehicleType],
  vin: Option[String],
  licensePlateNumber: Option[String],
  externalId: Option[String],
  hub: Option[ServiceHub],
  programs: Option[List[Program]],
  seatQuantity: Option[Long],
  picture: Option[Asset],
  markerIcon: Option[Asset],
  trackingColor: Option[String],
  belongsTo: Option[Account],
  zone: Option[Region],
  notes: Option[List[Note]],
  noteCount: Option[Long],
  inUse: Option[Boolean],
  contentName: Option[String]
)

object Vehicle {
  implicit lazy val vehicleJsonFormat: Format[Vehicle] = Json.format[Vehicle]
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/Vehicle.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class Vehicle 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public VehicleType VehicleType { get; set; }
    public string Vin { get; set; }
    public string LicensePlateNumber { get; set; }
    public string ExternalId { get; set; }
    public ServiceHub Hub { get; set; }
    public List<Program> Programs { get; set; }
    public long SeatQuantity { get; set; }
    public Asset Picture { get; set; }
    public Asset MarkerIcon { get; set; }
    public string TrackingColor { get; set; }
    public Account BelongsTo { get; set; }
    public Region Zone { get; set; }
    public List<Note> Notes { get; set; }
    public long NoteCount { get; set; }
    public bool InUse { get; set; }
    public string ContentName { get; set; }
}


